sktime/forecasting/compose/_skforecast_reduce.py

```diff
@@ -0,0 +1,445 @@
+
+    def _make_index_compatible(self, df, input_var):
+        if df is None:
+            return None
+
+        new_df = df
+        if not isinstance(df.index, (pd.RangeIndex, pd.DatetimeIndex)):
+            if isinstance(df.index, pd.PeriodIndex):
+                new_df = self._coerce_period_to_datetime_index(df)
+            elif pd.api.types.is_integer_dtype(df.index):
+                try:
+                    new_df = self._coerce_int_to_range_index(df)
+                except AssertionError:
+                    raise ValueError(
+                        f"Coercion of index of {input_var} from integer pd.Index to "
+                        "pd.RangeIndex failed. Please ensure that indexes are equally "
+                        "spaced apart."
+                    )
+            else:
+                raise ValueError(
+                    f"{input_var} must have one of the following index types: "
+                    "pd.RangeIndex, pd.DatetimeIndex, pd.PeriodIndex, pd.Index"
+                    f"(int dtype). Found index of type: {type(df.index)}"
+                )
+        return new_df
+
+    def _fit(
+        self: "SkforecastAutoreg",
+        y: pd.Series,
+        X: Optional[pd.DataFrame],
+        fh: Optional[ForecastingHorizon],
+    ):
+        """Fit forecaster to training data.
+
+        private _fit containing the core logic, called from fit
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        y : pd.Series
+            Target time series to which to fit the forecaster.
+        fh : guaranteed to be ForecastingHorizon or None, optional (default=None)
+            The forecasting horizon with the steps ahead to to predict.
+            Required (non-optional) here.
+        X : pd.DataFrame, optional (default=None)
+            Exogeneous time series to fit to.
+
+        Returns
+        -------
+        self : reference to self
+        """
+        del fh  # avoid being detected as unused by ``vulture`` like tools
+
+        self._forecaster = self._create_forecaster()
+
+        # skforecast does not support PeriodIndex and Integer Index.
+        # So converting to supported index types here if necessary.
+        y_new = self._make_index_compatible(y, "y")
+        X_new = self._make_index_compatible(X, "X")
+
+        self._forecaster.fit(y_new, exog=self._coerce_column_names(X_new))
+
+        return self
+
+    def _get_horizon_details(
+        self: "SkforecastAutoreg", fh: Optional[ForecastingHorizon]
+    ):
+        if not fh.is_all_out_of_sample(self.cutoff):
+            raise NotImplementedError(
+                f"{self.__class__.__name__} does not support in-sample predictions."
+            )
+
+        if not (out_of_sample_horizon := fh.to_out_of_sample(self.cutoff)):
+            raise ValueError(
+                f"{self.__class__.__name__} received empty out-of-sample horizon."
+            )
```

Comment:
This part was written by you actually. I am not very familiar with fh, but I think it does look like a redundant check. The previous if statement ensures that fh is 'all out of sample' so `out_of_sample_horizon` cannot never be None here. Is this correct?

Reply:
I also felt it was redundant, don't remember why I wrote that. Let me see my local branch internal commit history.

Reply:
Is it enough reason to keep this check?

```python
>>> 
>>> ForecastingHorizon([]).is_all_in_sample()
True
>>> ForecastingHorizon([]).is_all_out_of_sample()
True
>>> 
```

I am not too convinced, but will probably prefer to keep as it is since it's not doing any harm.

Reply:
I did actually test this case, I found that sktime doesn't allow empty fh to be passed to the estimator and error is raised beforehand itself. 
So I would say it's safe to remove.